src/TODOApp.js

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import TODOAddForm from './TODOAddForm';
+import TODOList from './TODOList';
+import AppHead from './AppHead';
+import './App.css';
+
+class TODOApp extends React.Component {
+  constructor(props){
+    super(props);
+    this.state = {
+      data: this.props.tasks
+    };
+  }
+
+  handleTaskRemoval = (taskId) => {
+		let data = this.state.data;
+		data = data.filter(function (e) {
+			return e.id !== taskId;
+		});
+		this.setState({data : data});
+	}
+
+  handleTaskUpdation = (task, taskId) => {
+		let data = this.state.data;
+    data[data.findIndex((obj => obj.id === taskId))].task = task;
+		this.setState({data : data});
+	}
+
+  handleSubmit = (task) => {
+		let data = this.state.data;
+		let id = data[data.length - 1].id + 1;
```

Comment:
Since ID is supposed to be unique every time. 
The logic above will create same ID for some tasks.
Please use some ID generator method for this logic.

Reply:
the id generated will be unique always as 'id' will take the value of last task's id and increment it by 1

Reply:
Ok, so how first task is created? what will be its id?

Reply:
This will be an issue if there are no tasks initially. I will put a check on task length

Reply:
I will still recommend to use some ID generator method to generate unique IDs for you. 
This will make your logic more simpler and if-else free, which we always look for.

Reply:
You can use or refer: https://gist.github.com/gordonbrander/2230317